nomenclature/codelist.py

```diff
@@ -788,6 +788,16 @@ def from_directory(
                     )
                 )
             mapping[code.name] = code
+
+        for code in list(mapping):
```

Comment:
To me this does not belong into the constructor of the class but rather is a classic application of a pydantic validator.
I have built this on the branch so if you don't mind, I'd push my changes.

Reply:
Yes, please push, thank you!